frontend/packages/topology/src/utils/application-utils.ts

```diff
@@ -136,29 +136,36 @@ const safeKill = async (model: K8sKind, obj: K8sResourceKind) => {
   return null;
 };
 
-const deleteWebhooks = (
+const deleteWebhooks = async (
   resource: K8sResourceKind,
   buildConfigs: K8sResourceKind[],
   isKnativeResource: boolean,
 ) => {
   const deploymentsAnnotations = resource.metadata?.annotations ?? {};
   const gitType = detectGitType(deploymentsAnnotations['app.openshift.io/vcs-uri']);
+  const secretList = await k8sList(SecretModel, {
+    ns: resource.metadata.namespace,
+  });
   return buildConfigs?.reduce((requests, bc) => {
     const triggers = bc.spec?.triggers ?? [];
     const reqs = triggers.reduce((a, t) => {
       let obj: K8sResourceKind;
       const webhookType = t.generic ? 'generic' : gitType;
       const webhookTypeObj = t.generic || (!isKnativeResource && t[gitType]);
       if (webhookTypeObj) {
-        obj = {
-          ...resource,
-          metadata: {
-            name:
-              webhookTypeObj.secretReference?.name ??
-              `${resource.metadata.name}-${webhookType}-webhook-secret`,
-            namespace: resource.metadata.namespace,
-          },
-        };
+        const secretName =
+          webhookTypeObj.secretReference?.name ??
+          `${resource.metadata.name}-${webhookType}-webhook-secret`;
+        const secretResource = secretList.find((secret) => secret.metadata.name === secretName);
+        if (!_.isEmpty(secretResource)) {
+          obj = {
+            ...resource,
+            metadata: {
+              name: secretName,
+              namespace: resource.metadata.namespace,
+            },
+          };
+        }
       }
       return obj ? [...a, safeKill(SecretModel, obj)] : a;
```

Comment:
Also since here we are now getting the list of the secrets in the ns and finding if the secret we want to delete exists and then deleting it, why not use `secretResource` here rather than creating another `obj`?